### ISSUE TITLE
Installing via Helm reports ElasticSearch discovery errors

### ISSUE THREAD

**Issue by AustinBGibbons** [amundsen-io/amundsen#830]
Hi! We are trying to test Amundsen using the helm templates to deploy. We referenced the README at https://github.com/amundsen-io/amundsen/tree/master/amundsen-kube-helm, but ran into issues with the ElasticSearch cluster.

## Expected Behavior
We took the following steps:

Download the Amundsen helm templates:

```
$ git clone git@github.com:plaid/amundsen.git
$ cp -r newmundsen/amundsen-kube-helm/ amundsen/
```

Configure dependencies

```
$ helm repo add elasticsearch https://kubernetes-charts.storage.googleapis.com
$ helm dependency update ./templates/helm/
```

installing, with no modifications to values.yml

```
$ helm upgrade --install amundsen ./templates/helm --values templates/helm/values.yaml --kube-context=<ctx>   --namespace=<ns>
Release "amundsen" does not exist. Installing it now.
NAME: amundsen
LAST DEPLOYED: Tue Dec  1 18:28:31 2020
NAMESPACE: <ns>
STATUS: deployed
REVISION: 1
```

## Current Behavior

Some of the pods start up:

```
$ kubectl -n <ns> get pods | grep amundsen
amundsen-elasticsearch-client-58996dd768-tsfbs   0/1     Running   0          6m26s
amundsen-elasticsearch-data-0                    1/1     Running   0          6m26s
amundsen-elasticsearch-master-0                  1/1     Running   0          6m26s
amundsen-frontend-79d647b9cf-n86gm               1/1     Running   0          6m26s
amundsen-metadata-6466556944-ntg2q               1/1     Running   0          6m26s
amundsen-search-b77655576-9nbvf                  1/1     Running   0          6m26s
```

but, with logging errors:

elasticsearch-client

```
...
$ kubectl -n <ns> logs amundsen-elasticsearch-client-58996dd768-tsfbs

[2020-12-02T02:35:42,911][WARN ][o.e.d.z.ZenDiscovery     ] [amundsen-elasticsearch-client-58996dd768-tsfbs] not enough master nodes discovered during pinging (found [[]], but needed [1]), pinging again
[2020-12-02T02:35:43,071][WARN ][r.suppressed             ] [amundsen-elasticsearch-client-58996dd768-tsfbs] path: /_cluster/health, params: {}
org.elasticsearch.discovery.MasterNotDiscoveredException: null
	at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction$4.onTimeout(TransportMasterNodeAction.java:262) [elasticsearch-6.8.2.jar:6.8.2]
	at org.elasticsearch.cluster.ClusterStateObserver$ContextPreservingListener.onTimeout(ClusterStateObserver.java:322) [elasticsearch-6.8.2.jar:6.8.2]
	at org.elasticsearch.cluster.ClusterStateObserver$ObserverClusterStateListener.onTimeout(ClusterStateObserver.java:249) [elasticsearch-6.8.2.jar:6.8.2]
	at org.elasticsearch.cluster.service.ClusterApplierService$NotifyTimeout.run(ClusterApplierService.java:564) [elasticsearch-6.8.2.jar:6.8.2]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:681) [elasticsearch-6.8.2.jar:6.8.2]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:835) [?:?]
[2020-12-02T02:35:45,912][WARN ][o.e.d.z.ZenDiscovery     ] [amundsen-elasticsearch-client-58996dd768-tsfbs] not enough master nodes discovered during pinging (found [[]], but needed [1]), pinging again
```

amundsen-elasticsearch-master-0

```
...
[2020-12-02T02:28:56,749][INFO ][o.e.p.PluginsService     ] [amundsen-elasticsearch-master-0] no plugins loaded
[2020-12-02T02:29:02,051][INFO ][o.e.d.DiscoveryModule    ] [amundsen-elasticsearch-master-0] using discovery type [zen] and host providers [settings]
[2020-12-02T02:29:03,350][INFO ][o.e.n.Node               ] [amundsen-elasticsearch-master-0] initialized
[2020-12-02T02:29:03,350][INFO ][o.e.n.Node               ] [amundsen-elasticsearch-master-0] starting ...
[2020-12-02T02:29:03,673][INFO ][o.e.t.TransportService   ] [amundsen-elasticsearch-master-0] publish_address {100.64.25.30:9300}, bound_addresses {0.0.0.0:9300}
[2020-12-02T02:29:03,730][INFO ][o.e.b.BootstrapChecks    ] [amundsen-elasticsearch-master-0] bound or publishing to a non-loopback address, enforcing bootstrap checks
[2020-12-02T02:29:03,784][WARN ][o.e.d.z.UnicastZenPing   ] [amundsen-elasticsearch-master-0] failed to resolve host [amundsen-elasticsearch-discovery]
java.net.UnknownHostException: amundsen-elasticsearch-discovery: Name or service not known
	at java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method) ~[?:?]
	at java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:929) ~[?:?]
	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1515) ~[?:?]
	at java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:848) ~[?:?]
	at java.net.InetAddress.getAllByName0(InetAddress.java:1505) ~[?:?]
	at java.net.InetAddress.getAllByName(InetAddress.java:1364) ~[?:?]
	at java.net.InetAddress.getAllByName(InetAddress.java:1298) ~[?:?]
	at org.elasticsearch.transport.TcpTransport.parse(TcpTransport.java:536) ~[elasticsearch-6.8.2.jar:6.8.2]
	at org.elasticsearch.transport.TcpTransport.addressesFromString(TcpTransport.java:489) ~[elasticsearch-6.8.2.jar:6.8.2]
	at org.elasticsearch.transport.TransportService.addressesFromString(TransportService.java:743) ~[elasticsearch-6.8.2.jar:6.8.2]
	at org.elasticsearch.discovery.zen.UnicastZenPing.lambda$resolveHostsLists$0(UnicastZenPing.java:189) ~[elasticsearch-6.8.2.jar:6.8.2]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:681) [elasticsearch-6.8.2.jar:6.8.2]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:835) [?:?]
[2020-12-02T02:29:06,852][INFO ][o.e.c.s.MasterService    ] [amundsen-elasticsearch-master-0] zen-disco-elected-as-master ([0] nodes joined), reason: new_master {amundsen-elasticsearch-master-0}{HaWRmNmrRI-AhuHfNw-rrg}{TkGCWhNwRPajFSZgYlGeag}{100.64.25.30}{100.64.25.30:9300}
[2020-12-02T02:29:06,856][INFO ][o.e.c.s.ClusterApplierService] [amundsen-elasticsearch-master-0] new_master {amundsen-elasticsearch-master-0}{HaWRmNmrRI-AhuHfNw-rrg}{TkGCWhNwRPajFSZgYlGeag}{100.64.25.30}{100.64.25.30:9300}, reason: apply cluster state (from master [master {amundsen-elasticsearch-master-0}{HaWRmNmrRI-AhuHfNw-rrg}{TkGCWhNwRPajFSZgYlGeag}{100.64.25.30}{100.64.25.30:9300} committed version [1] source [zen-disco-elected-as-master ([0] nodes joined)]])
[2020-12-02T02:29:06,867][INFO ][o.e.h.n.Netty4HttpServerTransport] [amundsen-elasticsearch-master-0] publish_address {100.64.25.30:9200}, bound_addresses {0.0.0.0:9200}
[2020-12-02T02:29:06,867][INFO ][o.e.n.Node               ] [amundsen-elasticsearch-master-0] started
```

We are able to hit the frontend, where we can see that the search does seem to be failing

![image](https://user-images.githubusercontent.com/1144402/100821273-e156ae80-3404-11eb-86d4-c415caa8ced2.png)

## Version info

Amundsen Master from 12/1/2020

```
$ helm version
version.BuildInfo{Version:"v3.2.0", GitCommit:"e11b7ce3b12db2941e90399e874513fbd24bcb71", GitTreeState:"clean", GoVersion:"go1.13.10"}
```

We did not modify the elasticsearch version, so we think we should be using 1.32.0

## Context
We're trying to test it out, but aren't sure how to debug helm->ES issues

## Your Environment
The k8s cluster was built about a year ago (late 2019 / early 2020)

**Comment by feng-tao** [comment]
cc @javamonkey79 
But I would suggest to use a managed ES service instead of creating one yourself. It is a bit complicated to build/operate one (https://github.com/pires/kubernetes-elasticsearch-cluster)

**Comment by feng-tao** [comment]
another thing worth mentioning once your current issue is solved is to make sure ES is within 6.x(we test 6.7).

**Comment by AustinBGibbons** [comment]
Thanks for the reply @feng-tao , I also tried with 

```
helm repo add elasticsearch https://helm.elastic.co/
```

but seem to have the same issue. By managed ES service, do you mean something like AWS managed ElasticSearch? I will try that as well. 

To do that, is it correct to change enabled to false here:

```
elasticsearch:
  # elasticsearch.enabled -- set this to false, if you want to provide your own ES instance.
  enabled: false
```

and then  modify teh search service here

```
  ##
  ## search.elasticsearchEndpoint -- The name of the service hosting elasticsearch on your cluster, if you bring your own. You should only need to change this, if you don't use the version in this chart.
  ##
  elasticsearchEndpoint:
```

to point to an ES URL?

```
elasticsearchEndpoint: https://<cluster>.us-east-1.es.amazonaws.com
```

**Comment by javamonkey79** [comment]
@AustinBGibbons we run ES in our chart and not standalone. This is all we have in our values file, to make it work:
```
elasticsearch:
  client:
    serviceType: LoadBalancer
    serviceAnnotations:
      external-dns.alpha.kubernetes.io/hostname: amundsen-elasticsearch-someenvironment.some-account-name.accounts.my-company.com
      service.beta.kubernetes.io/aws-load-balancer-internal: 0.0.0.0/0
      service.beta.kubernetes.io/aws-load-balancer-type: nlb
```

I checked my shell history, and we did not have to do this:
```
$ helm repo add elasticsearch https://kubernetes-charts.storage.googleapis.com
$ helm dependency update ./templates/helm/
```

The thing that sticks out to me, is this:
```
$ helm version
version.BuildInfo{Version:"v3.2.0", GitCommit:"e11b7ce3b12db2941e90399e874513fbd24bcb71", GitTreeState:"clean", GoVersion:"go1.13.10"}
```

Per the readme, here is the helm prerequisite:
Helm 2.14+

However, that "+" is a little misleading. I think it should be:
Helm [2,3)

tl;dr; this chart needs to be on helm v2 even better v2.14. I suspect that with helm 3 it is not pulling the right ES dependent chart, or, in the wrong order. I think your choices are to:

1. Try with helm 2 (I understand this may not be possible for you)
2. Try with a managed instance - it looks like you have made correct assumptions in your previous comment
3. Try to narrow down which ES chart helm 3 is installing, and try to get it to the right version; if that is already they case, then there is likely some unexpected config; based on your original log post, it looks like the amundsen-elasticsearch-discovery is either not being created, or being created out of order; I'm not sure. 
4. Try updating your values file, to create a loadbalanced client as I shared above, it might be why the discovery service isn't coming up.

**Comment by AustinBGibbons** [comment]
Thanks for the reply @javamonkey79 , yeah unfortunately we're locked into helm 3.x - I am also getting errors with the frontend service trying to communicate with the metadata service as well as some AUTH configurations I don't understand, so perhaps it's just incompatible - I am a helm newbie so not sure how to say for sure. I will experiment restarting from scratch on helm 3, thanks!

(frontend -> metadata errors when hitting the landing page, fwiw)

```
2020-12-02T16:05:43+0000.541 [ERROR] v0.current_user:47 (9:MainThread) - Encountered exception: AUTH_USER_METHOD is not configured
Traceback (most recent call last):
  File "/app/amundsen_application/api/v0.py", line 28, in current_user
    raise Exception('AUTH_USER_METHOD is not configured')
Exception: AUTH_USER_METHOD is not configured
2020-12-02T16:05:43+0000.551 [ERROR] v0.get_bookmark:517 (8:MainThread) - Encountered exception: AUTH_USER_METHOD is not configured
Traceback (most recent call last):
  File "/app/amundsen_application/api/metadata/v0.py", line 491, in get_bookmark
    raise Exception('AUTH_USER_METHOD is not configured')
Exception: AUTH_USER_METHOD is not configured
2020-12-02T16:05:43+0000.560 [DEBUG] connectionpool._new_conn:205 (10:MainThread) - Starting new HTTP connection (1): amundsen-metadata:5002
2020-12-02T16:05:43+0000.979 [DEBUG] connectionpool._new_conn:205 (11:MainThread) - Starting new HTTP connection (1): amundsen-metadata:5002
2020-12-02T16:05:44+0000.404 [DEBUG] connectionpool._new_conn:205 (9:MainThread) - Starting new HTTP connection (1): amundsen-metadata:5002
2020-12-02T16:05:46+0000.565 [ERROR] v0.get_tags:363 (10:MainThread) - Encountered exception: HTTPConnectionPool(host='amundsen-metadata', port=5002): Max retries exceeded with url: /tags/ (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7fad64b839d0>, 'Connection to amundsen-metadata timed out. (connect timeout=3)'))
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 159, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw)
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 80, in create_connection
    raise err
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 70, in create_connection
    sock.connect(sa)
socket.timeout: timed out
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 354, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.7/http/client.py", line 1277, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1323, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1272, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1032, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.7/http/client.py", line 972, in send
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 181, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 164, in _new_conn
    (self.host, self.timeout))
urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPConnection object at 0x7fad64b57c10>, 'Connection to amundsen-metadata timed out. (connect timeout=3)')
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 399, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='amundsen-metadata', port=5002): Max retries exceeded with url: /popular_tables/?limit=4 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7fad64b57c10>, 'Connection to amundsen-metadata timed out. (connect timeout=3)'))
During handling of the above exception, another exception occurred:

...
```

**Comment by javamonkey79** [comment]
Thanks for the update @AustinBGibbons, best of luck.

@feng-tao I would vote that this issue be closed on account of the helm 2 vs 3 issue and that we put up a story to move to helm 3. My team and I won't be able to work on it until at least Q1, if at all, but, there may be others who are interested in doing the work.

**Comment by feng-tao** [comment]
sg

**Comment by jornh** [comment]
@AustinBGibbons is your Neo4j pod up and alive? Do you have data loaded into it as per step 6 in https://www.amundsen.io/amundsen/installation/#bootstrap-a-default-version-of-amundsen-using-docker

Also, as long as you haven’t configured authentication the `AUTH_USER_METHOD` is expected.

`2020-12-02T16:05:46+0000.565 [ERROR] v0.get_tags:363 ...` in https://github.com/amundsen-io/amundsen/issues/830#issuecomment-737409848 on the metadata service looks to me like your main issue

- - - 

For the record we’re happily using helm 3 here - and have been since https://github.com/amundsen-io/amundsen/pull/373 (although not on latest Amundsen chart version, but something around chart version 1.1’ish)

Another indication of the same (see issue title): https://github.com/amundsen-io/amundsen/issues/486

Longer term we should switch to official ES chart mentioned first in https://github.com/amundsen-io/amundsen/issues/830#issuecomment-737320815 - the old chart, even the whole helm `stable` repo is deprecated. It AFAIK was even taken down in November but has been put back up because it broke to much for people. See October 26th post and later on https://helm.sh/blog/

**Comment by AustinBGibbons** [comment]
I think all of this was configuration specific to our k8s setup, so going to close this out, thank you!
